File: Flash Sort/FlashSort.java
class FlashSort {

    static void flashSort(int arr[]) {

        int n = arr.length;

        int min = arr[0];
        int maxIndex = 0;

        for(int i = 1; i < n; i++) {

            if(arr[i] < min)
                min = arr[i];

            if(arr[i] > arr[maxIndex])
                maxIndex = i;
        }

        if(arr[maxIndex] == min)
            return;

        for(int i = 1; i < n; i++) {

            int key = arr[i];
            int j = i - 1;

            while(j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j--;
            }

            arr[j + 1] = key;
        }
    }

    public static void main(String[] args) {

        int arr[] = {30, 20, 10, 50, 40};

        flashSort(arr);

        for(int num : arr)
            System.out.print(num + " ");
    }
}

File: Flash Sort/FlashSort.py
def flash_sort(arr):

    n = len(arr)

    if n <= 1:
        return

    minimum = min(arr)
    maximum = max(arr)

    if minimum == maximum:
        return

    # Final insertion sort phase
    for i in range(1, n):

        key = arr[i]
        j = i - 1

        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1

        arr[j + 1] = key

arr = [30, 20, 10, 50, 40]

flash_sort(arr)

print(arr)
